frontend/app/src/components/hexagonalBoard/HexagonalBoard.tsx

```diff
@@ -0,0 +1,113 @@
+import {
+	useState,
+	ReactNode,
+	useEffect,
+	forwardRef,
+	useImperativeHandle,
+	ForwardedRef,
+} from "react";
+import HexagonalTile, {
+	HexagonalTileProps,
+} from "@/components/hexagonalTile/HexagonalTile.tsx";
+import "./HexagonalBoard.css";
+
+const defaultTileProps: HexagonalTileProps = {
+	fill: "",
+	isDimmed: false,
+	stroke: "gray",
+	strokeWidth: "16",
+	identifier: "",
+	textContent: "",
+};
+
+interface HexagonalBoardProps {
+	rows: number;
+	tilesPerRow: number;
+}
+
+const HexagonalBoard = forwardRef(function HexagonalBoard(
+	{ rows, tilesPerRow }: HexagonalBoardProps,
+	ref?: ForwardedRef<{
```

Comment:
I think you previously declared the type of the ref in the interface but didn't use it, why didn't you leave it their and just destructured it from the props in the function header?

Reply:
Because, as I said in my previous comment,

> You need to pass the ref separately in the forwardRef syntax, so I moved the type into the parameter.

The ref is **NOT** part of the props for the render function when using forwardRef.

See https://react.dev/reference/react/forwardRef#exposing-an-imperative-handle-instead-of-a-dom-node for more details.

---

Correct me if I am wrong, I may misunderstand forwardRef.

Reply:
what previous comment?sry couldn't find it, but alrigth :+1: